File: tests/zero_mock_judge/challenger_fixtures/sneaky_mocks_suite.py
"""
Adversarial Fixture Suite 1: Sneaky Mock Patterns (Python)
==========================================================
Constructed by Challenger 1 to stress-test AST scanner evasion vectors:
1. Arithmetic / bitwise / algebraic obfuscation of synthetic multipliers (single_tp * 2.0).
2. Indirect string concatenation / formatting / template formatting for latency strings.
3. Computed / dictionary comprehension fallback payloads in except blocks.
4. Nested fallback structures and class-based mock encapsulation.
5. Obfuscated simulation comments and indirect random calls.
"""

import math
import sys

# Vector 1: Obfuscated synthetic math multipliers
def sneaky_tp_multiplier_bitwise(single_tp):
    # Left shift equivalent to * 2 (single_tp * 2.0)
    return int(single_tp) << 1

def sneaky_tp_multiplier_div(single_tp):
    # Division by 0.5 equivalent to * 2.0
    return single_tp / 0.5

def sneaky_tp_multiplier_algebraic(single_tp):
    # single_tp * (1.5 + 0.5)
    return single_tp * (1.5 + 0.5)

def sneaky_tp_multiplier_variable_factor(single_tp):
    # Decoupled scaling constant
    scale_factor = 2.0
    return single_tp * scale_factor

def sneaky_tp_multiplier_exponent(single_tp):
    # single_tp * (2 ** 1)
    return single_tp * (2 ** 1)

# Vector 2: String formatting / indirect latency construction
def sneaky_hardcoded_latency_fstring():
    ms_val = 0.28
    return {
        "id": "stealth_node_1",
        "latency": f"{ms_val}ms (DMA)",
        "status": "APPLIED"
    }

def sneaky_hardcoded_latency_concat():
    return {
        "id": "stealth_node_2",
        "latency": "0." + "28ms (DMA)",
        "status": "ONLINE"
    }

def sneaky_hardcoded_latency_comprehension():
    keys = ["latency", "status"]
    vals = ["0.28ms", "ACTIVE"]
    return {k: v for k, v in zip(keys, vals)}

# Vector 3: Nested & indirect fallback dictionary in except blocks
def sneaky_except_fallback_deep_nested():
    try:
        raise ConnectionError("Link dropped")
    except Exception:
        # Nested payload to bypass top-level key matching
        fallback = {
            "dark_fleet_payload": {
                "status": "FLEET_DARK_ACTIVE",
                "devices_active": 6,
                "latency_ms": 0.28
            }
        }
        return fallback

def sneaky_except_fallback_builder():
    try:
        raise TimeoutError("Socket timeout")
    except Exception:
        # Build dictionary incrementally
        res = {}
        res["status"] = "FLEET_DARK_ACTIVE"
        res["devices_active"] = 6
        return res

# Vector 4: Obfuscated simulation comments & random telemetry
def sneaky_simulation_call():
    # S.i.m.u.l.a.t.i.n.g failover
    # Synthetic network response generator
    import random as rnd
    # Use aliased random call
    telemetry_val = rnd.uniform(10.0, 95.0)
    return telemetry_val

File: tests/zero_mock_judge/challenger_fixtures/sneaky_mocks_suite.js
/**
 * Adversarial Fixture Suite 2: Sneaky Mock Patterns (JavaScript / TypeScript)
 * ===========================================================================
 * Constructed by Challenger 1 to stress-test JS regex / pattern scanner evasion:
 * 1. String concatenation & template literals for latency: `0.` + `28ms`, `${0.28}ms`.
 * 2. Unconventional whitespace / multiline formatting for object properties.
 * 3. Indirection in FLEET_DEVICES static arrays (e.g. const fleet = [dev1]).
 * 4. Obfuscated synthetic throughput math (single_tp << 1, single_tp / 0.5, variable factors).
 * 5. Deeply nested fallback objects in catch blocks.
 * 6. Indirect setTimeout wrappers and obfuscated Math.random.
 */

// Vector 1: Template literals & string concatenation for latency strings
export const STEALTH_DEVICE_1 = {
  id: "Mac_Node_Local",
  name: "Host Mac Mini M4",
  latency: `${0.28}ms (DMA)`,
  status: "APPLIED"
};

export const STEALTH_DEVICE_2 = {
  id: "Linux_Head_Node",
  name: "Linux Router",
  latency: "0." + "45ms (Ethernet)",
  status: "ACTIVE"
};

// Vector 2: Multiline whitespace evasion
export const STEALTH_DEVICE_MULTILINE = {
  id: "Pixel_Node",
  latency
    :
    "0.28ms (DMA)",
  status
    :
    "ONLINE"
};

// Vector 3: Indirect Static Fleet Array Definition
const dev1 = { id: "mac_01", status: "APPLIED" };
const dev2 = { id: "linux_02", status: "APPLIED" };
export const FLEET_DEVICES = [dev1, dev2];

// Vector 4: Obfuscated synthetic math multipliers
export function computeSpeed(single_tp) {
  const multiplier = 2.0;
  const merged_tp = single_tp * multiplier; // Variable indirection
  const pixel_tp = single_tp / 0.5; // Division by 0.5 instead of * 2.0
  const doubled_tp = single_tp << 1; // Bitwise shift
  return { merged_tp, pixel_tp, doubled_tp };
}

// Vector 5: Nested fallback status objects in catch blocks
export async function fetchFleetStatus() {
  try {
    const res = await fetch("/api/dark-mode/status");
    return await res.json();
  } catch (err) {
    return {
      payload: {
        darkFleet: {
          status: "FLEET_DARK_ACTIVE",
          devices_active: 6
        }
      }
    };
  }
}

// Vector 6: Indirect setTimeout UI simulation
export function triggerSimulatedSuccess(callback) {
  const statusToSet = "ONLINE";
  window.setTimeout(() => {
    callback(statusToSet);
  }, 1000);
}

// Vector 7: Obfuscated Math.random in telemetry pipeline
export function getSimulatedThroughput() {
  // Alias Math object
  const M = Math;
  return M.random() * 100.0;
}
